### PR TITLE
Allow setting multi-universe fills

### LINKED ISSUE #714: Multi-Universe fill can't be created from scratch
**Describe the bug**

Multi-universe fills can't be realistically updated, and is really just read-only

**To Reproduce**

``` python
import montepy
cell = montepy.Cell(number=1)
lattice_cell.fill.multiple_universes = True
inst_universe = montepy.Universe(1)
guide_universe = montepy.Universe(2)
lattice_cell.fill.universes = np.array([[inst_universe, guide_universe]])
lattice_cell.format_for_mcnp_input((6,2,0))
```

**Error Message (if any)**

This leads to:

<details>

``` python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[50], line 1
----> 1 lattice_cell.format_for_mcnp_input((6,2,0))

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:50](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=49), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     48 self._handling_exception = True
     49 try:
---> 50     add_line_number_to_exception(e, self)
     51 finally:
     52     del self._handling_exception

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py:225](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py#line=224), in add_line_number_to_exception(error, broken_robot)
    223 # avoid calling this n times recursively
    224 if hasattr(error, "montepy_handled"):
--> 225     raise error
    226 error.montepy_handled = True
    227 args = error.args

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:42](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=41), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     39 @functools.wraps(func)
     40 def wrapped(*args, **kwargs):
     41     try:
---> 42         return func(*args, **kwargs)
     43     except Exception as e:
     44         if len(args) > 0 and isinstance(args[0], MCNP_Object):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/cell.py:742](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/cell.py#line=741), in Cell.format_for_mcnp_input(self, mcnp_version)
    728 """Creates a string representation of this MCNP_Object that can be
    729 written to file.
    730 
   (...)    739     a list of strings for the lines that this input will occupy.
    740 """
    741 self.validate()
--> 742 self._update_values()
    743 self._tree.check_for_graveyard_comments()
    744 modifier_keywords = {
    745     cls._class_prefix(): cls for cls in self._INPUTS_TO_PROPERTY.keys()
    746 }

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:50](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=49), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     48 self._handling_exception = True
     49 try:
---> 50     add_line_number_to_exception(e, self)
     51 finally:
     52     del self._handling_exception

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py:225](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py#line=224), in add_line_number_to_exception(error, broken_robot)
    223 # avoid calling this n times recursively
    224 if hasattr(error, "montepy_handled"):
--> 225     raise error
    226 error.montepy_handled = True
    227 args = error.args

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:42](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=41), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     39 @functools.wraps(func)
     40 def wrapped(*args, **kwargs):
     41     try:
---> 42         return func(*args, **kwargs)
     43     except Exception as e:
     44         if len(args) > 0 and isinstance(args[0], MCNP_Object):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/cell.py:639](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/cell.py#line=638), in Cell._update_values(self)
    637 self._tree.nodes["geometry"] = self.geometry.node
    638 for input_class, (attr, _) in self._INPUTS_TO_PROPERTY.items():
--> 639     getattr(self, attr)._update_values()

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:50](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=49), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     48 self._handling_exception = True
     49 try:
---> 50     add_line_number_to_exception(e, self)
     51 finally:
     52     del self._handling_exception

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py:225](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py#line=224), in add_line_number_to_exception(error, broken_robot)
    223 # avoid calling this n times recursively
    224 if hasattr(error, "montepy_handled"):
--> 225     raise error
    226 error.montepy_handled = True
    227 args = error.args

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:42](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=41), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     39 @functools.wraps(func)
     40 def wrapped(*args, **kwargs):
     41     try:
---> 42         return func(*args, **kwargs)
     43     except Exception as e:
     44         if len(args) > 0 and isinstance(args[0], MCNP_Object):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/cell_modifier.py:224](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/cell_modifier.py#line=223), in CellModifierInput._update_values(self)
    222 def _update_values(self):
    223     if self.in_cell_block:
--> 224         self._update_cell_values()
    225     else:
    226         new_vals = self._collect_new_values()

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:50](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=49), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     48 self._handling_exception = True
     49 try:
---> 50     add_line_number_to_exception(e, self)
     51 finally:
     52     del self._handling_exception

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py:225](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py#line=224), in add_line_number_to_exception(error, broken_robot)
    223 # avoid calling this n times recursively
    224 if hasattr(error, "montepy_handled"):
--> 225     raise error
    226 error.montepy_handled = True
    227 args = error.args

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:42](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=41), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     39 @functools.wraps(func)
     40 def wrapped(*args, **kwargs):
     41     try:
---> 42         return func(*args, **kwargs)
     43     except Exception as e:
     44         if len(args) > 0 and isinstance(args[0], MCNP_Object):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py:509](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py#line=508), in Fill._update_cell_values(self)
    507 else:
    508     self._tree["classifier"].modifier = None
--> 509 new_vals = self._update_cell_universes(new_vals)
    510 if self.transform is None:
    511     try:

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:50](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=49), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     48 self._handling_exception = True
     49 try:
---> 50     add_line_number_to_exception(e, self)
     51 finally:
     52     del self._handling_exception

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py:225](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py#line=224), in add_line_number_to_exception(error, broken_robot)
    223 # avoid calling this n times recursively
    224 if hasattr(error, "montepy_handled"):
--> 225     raise error
    226 error.montepy_handled = True
    227 args = error.args

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:42](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=41), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     39 @functools.wraps(func)
     40 def wrapped(*args, **kwargs):
     41     try:
---> 42         return func(*args, **kwargs)
     43     except Exception as e:
     44         if len(args) > 0 and isinstance(args[0], MCNP_Object):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py:552](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py#line=551), in Fill._update_cell_universes(self, new_vals)
    550 if self.multiple_universes:
    551     payload = []
--> 552     for i in self._axis_range(0):
    553         for j in self._axis_range(1):
    554             for k in self._axis_range(2):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:50](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=49), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     48 self._handling_exception = True
     49 try:
---> 50     add_line_number_to_exception(e, self)
     51 finally:
     52     del self._handling_exception

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py:225](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py#line=224), in add_line_number_to_exception(error, broken_robot)
    223 # avoid calling this n times recursively
    224 if hasattr(error, "montepy_handled"):
--> 225     raise error
    226 error.montepy_handled = True
    227 args = error.args

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:42](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=41), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     39 @functools.wraps(func)
     40 def wrapped(*args, **kwargs):
     41     try:
---> 42         return func(*args, **kwargs)
     43     except Exception as e:
     44         if len(args) > 0 and isinstance(args[0], MCNP_Object):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py:461](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py#line=460), in Fill._axis_range(self, axis)
    448 def _axis_range(self, axis):
    449     """Returns an iterator for iterating over the given axis.
    450 
    451     Parameters
   (...)    459         range
    460     """
--> 461     return range(self._axis_size(axis))

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:50](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=49), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     48 self._handling_exception = True
     49 try:
---> 50     add_line_number_to_exception(e, self)
     51 finally:
     52     del self._handling_exception

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py:225](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py#line=224), in add_line_number_to_exception(error, broken_robot)
    223 # avoid calling this n times recursively
    224 if hasattr(error, "montepy_handled"):
--> 225     raise error
    226 error.montepy_handled = True
    227 args = error.args

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:42](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=41), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     39 @functools.wraps(func)
     40 def wrapped(*args, **kwargs):
     41     try:
---> 42         return func(*args, **kwargs)
     43     except Exception as e:
     44         if len(args) > 0 and isinstance(args[0], MCNP_Object):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py:476](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py#line=475), in Fill._axis_size(self, axis)
    463 def _axis_size(self, axis):
    464     """Get the length of the given axis.
    465 
    466     Parameters
   (...)    474         the length of the given axis of the universe matrix.
    475     """
--> 476     return int(self.max_index[axis] - self.min_index[axis]) + 1

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:50](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=49), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     48 self._handling_exception = True
     49 try:
---> 50     add_line_number_to_exception(e, self)
     51 finally:
     52     del self._handling_exception

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py:249](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/errors.py#line=248), in add_line_number_to_exception(error, broken_robot)
    247 args = (message,) + args[1:]
    248 error.args = args
--> 249 raise error.with_traceback(trace)

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py:42](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/mcnp_object.py#line=41), in _ExceptionContextAdder._wrap_attr_call.<locals>.wrapped(*args, **kwargs)
     39 @functools.wraps(func)
     40 def wrapped(*args, **kwargs):
     41     try:
---> 42         return func(*args, **kwargs)
     43     except Exception as e:
     44         if len(args) > 0 and isinstance(args[0], MCNP_Object):

File [~/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py:296](http://127.0.0.1:8888/home/mgale/mambaforge/envs/montepy_mc/lib/python3.12/site-packages/montepy/data_inputs/fill.py#line=295), in Fill.max_index(self)
    285 @property
    286 def max_index(self):
    287     """The maximum indices of the matrix in each dimension.
    288 
    289     For the order of the indices see: ``DIMENSIONS``.
   (...)    294         the maximum indices of the matrix for complex fills
    295     """
--> 296     return self._max_index

AttributeError: 'Fill' object has no attribute '_max_index'

Error came from Fill: Universe: None, transform: None from an unknown file.
```

</details>

**Version**

* version 1.0.0a4

### PR DESCRIPTION
**Pull request by MicahGale**
# Pull Request Checklist for MontePy

### Description

This fixes a "bug" where setting `Fill.universes` was not properly exported. There was previously... an effort to implement this, which was not successful. This changed:

1. Making sure `min_index` and `max_index` is always accessible without crashing
2. Setting `min_index` and `max_index` whenever `universes` is set.
3. setting `mulitple_universes=True` whenever `universes` is set. 
4. `_update_cell_values` works properly. 

Fixes #714

---

### General Checklist

- [x] I have performed a self-review of my own code.
- [x] The code follows the standards outlined in the [development documentation](https://idaholab.github.io/MontePy/developing.html).
- [x] I have formatted my code with `black` version 25.
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable).

---

<details open> 

<summary><h3>Documentation Checklist</h3></summary>

- [ ] I have documented all added classes and methods.
- [ ] For infrastructure updates, I have updated the developer's guide.
- [ ] For significant new features, I have added a section to the getting started guide.

</details>

---

<details>
<summary><h3>First-Time Contributor Checklist</h3></summary>

- [ ] If this is your first contribution, add yourself to `pyproject.toml` if you wish to do so.

</details>

---

### Additional Notes for Reviewers

Ensure that:

- [ ] The submitted code is consistent with the merge checklist outlined [here](https://www.montepy.org/developing.html#merge-checklist).
- [ ] The PR covers all relevant aspects according to the development guidelines.
- [ ] 100% coverage of the patch is achieved, or justification for a variance is given.

----
📚 Documentation preview 📚: https://montepy--717.org.readthedocs.build/en/717/

### CODE REVIEW

File: montepy/data_inputs/fill.py [hunk 1]
```diff
@@ -259,17 +271,32 @@ def universes(self):
     def universes(self, value):
         if not isinstance(value, (np.ndarray, type(None))):
             raise TypeError(f"Universes must be set to an array. {value} given.")
-        if not self.multiple_universes:
+        if len(value.shape) != 3:
```
Comment (tjlaboss) [inline montepy/data_inputs/fill.py:274]:
```python
a.ndim != 3
```

File: montepy/data_inputs/fill.py [hunk 2]
```diff
@@ -259,17 +271,32 @@ def universes(self):
...
             raise ValueError(
-                "Multiple universes can only be set when multiple_universes is True."
+                f"3D array must be given for fill.universes. Array of shape: {value.shape} given."
+            )
+        if value.dtype != np.object_ or any(
+            map(lambda x: not isinstance(x, (Universe, type(None))), value.flatten())
+        ):
```
Comment (tjlaboss) [inline montepy/data_inputs/fill.py:280]: This took me a second to understand. Thinking about it:

- Ideally, lattices are of the `Universe` type: consider leading with `not issubclass(value.dtype, Universe)`
- Otherwise, `value.dtype` can be `object`, not just  `np.object_`. 
- Use `value.flat` as an iterator, not `value.flatten()`
- Could equivalently use `not all(...lambda x: isinstance...`). (the current `any` is fine)

File: montepy/data_inputs/fill.py [hunk 3]
```diff
@@ -259,17 +271,32 @@ def universes(self):
...
+            raise TypeError(
+                f"All values in array must be a Universe (or None). {value} given."
             )
+        self.multiple_universes = True
+        if self.min_index is None:
+            self.min_index = np.array([0] * 3)
+        self.max_index = self.min_index + np.array(value.shape) - np.array([1, 1, 1])
```
Comment (tjlaboss) [inline montepy/data_inputs/fill.py:287]: instead of ` - np.array([1,1,1])`, just do ` - 1`.

File: montepy/data_inputs/fill.py [hunk 4]
```diff
@@ -259,17 +271,32 @@ def universes(self):
...
-                "Multiple universes can only be set when multiple_universes is True."
+                f"3D array must be given for fill.universes. Array of shape: {value.shape} given."
+            )
+        if value.dtype != np.object_ or any(
+            map(lambda x: not isinstance(x, (Universe, type(None))), value.flatten())
+        ):
+            raise TypeError(
+                f"All values in array must be a Universe (or None). {value} given."
             )
+        self.multiple_universes = True
```
Comment (tjlaboss) [inline montepy/data_inputs/fill.py:284]: Corner case: this will be `True` even if it's a 1x1x1 lattice, or another shape that's all `None` except for one lattice cell. Is the `True` setting here ok then?

File: tests/test_universe.py [hunk 5]
```diff
@@ -370,8 +374,10 @@ def test_fill_universes_setter(self):
         with self.assertRaises(TypeError):
             fill.universes = "hi"
         fill.multiple_universes = False
-        with self.assertRaises(ValueError):
-            fill.universes = fill_array
+        with pytest.raises(ValueError):
+            fill.universes = np.array([1, 2])
+        with pytest.raises(TypeError):
+            fill.universes = np.array([[[1]]])
```
Comment (tjlaboss) [inline tests/test_universe.py:380]: Isn't this technically a valid lattice?

File: montepy/data_inputs/fill.py [hunk 6]
```diff
@@ -579,7 +613,29 @@ def _value_node_generator():
...
+                )
+            return new_vals
+        vals_iter = iter(new_vals)
+        for min_idx, max_idx in zip(self.min_index, self.max_index):
+            min_val = next(vals_iter)
+            min_val.value = min_idx
+            next(vals_iter)
+            max_val = next(vals_iter)
+            max_val.value = max_idx
+        return new_vals
```
Comment (tjlaboss) [inline montepy/data_inputs/fill.py:641]: I'll need to look at this function again in more depth to understand it. Do you have an example of where it's relevant?

File: montepy/data_inputs/fill.py (hunk 1 above)
Comment (tjlaboss) [reply montepy/data_inputs/fill.py:274]: More generally:

Why enforce that `ndim` is 3? Shouldn't the user be allowed to represent a 2D lattice with a 2D array?

Comment (MicahGale) [reply montepy/data_inputs/fill.py:274]: My reading of section `5.5.5.3.1 Lattice Indexing` of the MCNP 6.3 manual seems to require all three dimensions, though it's implicit. Do you know of any documentation that is more explicit?

File: montepy/data_inputs/fill.py (hunk 2 above)
Comment (MicahGale) [reply montepy/data_inputs/fill.py:280]: My understanding is that `np.array([montepy.Universe(5)])` would only ever have the `dtype` `object` (or [`object_`](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.object_)), so I don't see how `issubclass(value.dtype, Universe)` would work here?

Comment (MicahGale) [reply montepy/data_inputs/fill.py:280]: But yeah I'll make a function to make this a function.

File: montepy/data_inputs/fill.py (hunk 4 above)
Comment (MicahGale) [reply montepy/data_inputs/fill.py:284]: `Fill= 0:0 0:0 0:0 1` is weird but valid. That's the goal of `multiple_universes` more how it represents data than what data are there.

File: tests/test_universe.py (hunk 5 above)
Comment (MicahGale) [reply tests/test_universe.py:380]: No because it's not a `montepy.Universe`

File: montepy/data_inputs/fill.py (hunk 6 above)
Comment (MicahGale) [reply montepy/data_inputs/fill.py:641]: It is relevant every time the multi-universe fill is used.

File: montepy/data_inputs/fill.py (hunk 2 above)
Comment (tjlaboss) [reply montepy/data_inputs/fill.py]: If the `dtype` is `Universe`, then you don't have to evaluate the `map`.

File: tests/test_universe.py (hunk 5 above)
Comment (tjlaboss) [reply tests/test_universe.py:380]: 🤦‍♀️

File: montepy/data_inputs/fill.py (hunk 1 above)
Comment (tjlaboss) [reply montepy/data_inputs/fill.py]: I'm implying that MontePy could add the extra dimension for the sake of convenience, probably at write time. Should the user have to representing a 2D lattice as a 3D array within Python?

Comment (MicahGale) [reply montepy/data_inputs/fill.py]: Ohh I see. I think there would be enough ambiguity that I think we should make that a feature request, and fully think it out there first.

File: montepy/data_inputs/fill.py (hunk 2 above)
Comment (MicahGale) [reply montepy/data_inputs/fill.py]: Right, but it looks like `dtype` can never be a universe. [`np.object_`](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.object_) is just a pyobject pointer.

File: montepy/data_inputs/fill.py (hunk 1 above)
Comment (tjlaboss) [reply montepy/data_inputs/fill.py]: Makes sense 👍

File: montepy/data_inputs/fill.py (hunk 2 above)
Comment (tjlaboss) [reply montepy/data_inputs/fill.py]: Right, I forgot that the `dtype` doesn't persist. I'll mark as resolved.

**Review by tjlaboss — APPROVED** [review]
